sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -3,27 +3,89 @@
 
 #pragma once
 
+#include <string>
+#include <vector>
+
 #include <internal/contract.hpp>
 
+using namespace std;
+
 namespace azure
 {
 namespace core
 {
 namespace http
 {
 
-class http_request
+namespace http_method
 {
+enum HttpMethod
+{
+  GET,
+  HEAD,
+  POST,
+  PUT,
+  DELETE,
+  PATCH,
+};
+}
 
+class Header
+{
 private:
-  /* data */
+  string name;
+  string value;
+
 public:
-  http_request();
-  ~http_request();
+  Header(string name, string value)
+  {
+    this->name = name;
+    this->value = value;
+  }
+  ~Header() {}
+  string getName();
+  string getValue();
 };
 
-http_request::http_request() {}
-http_request::~http_request() {}
+class Request
+{
+
+private:
+  http_method::HttpMethod method;
+  string url;
+  std::vector<Header> headers;
```

Comment:
>std::vector<Header> headers [](start = 2, length = 27)

Does this mean that two headers with same name may co-exist?

Reply:
will change it to a map

Reply:
I don't think it's a good idea to use `std::map`. In RFC2616 it says **it is "good practice" to send general-header fields first, followed by request-header or response- header fields, and ending with the entity-header fields.** So I think we should keep the order as it is.

If there are duplicate headers, then it's definitely a bug in client. In this case, we should find the bug and fix it, rather than try to cover it up with `std::map`.

Reply:
So, Golang uses map: https://golang.org/pkg/net/http/#Header
Java uses a Set for it: https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/HttpRequest.Builder.html#header(java.lang.String,java.lang.String)

So, I think we should use the map, as it looks people would expect no duplicates to be handle by the language itself

Reply:
> So, I think we should use the map, as it looks people would expect no duplicates to be handle by the language itself

okay, i'm fine with that.